Update typespec-ts generator references to Azure/typespec-azure (#38807)
Azure/azure-sdk-for-js: sdk/core/core-client/README.md

Reviewer: This package is not used for typespec-ts emitter, it probably is fine to link to the old repo since the old repo assuming that it would be archived but not deleted.

# Azure Core Service client library for JavaScript

> This library is primarily intended to be used in code generated by the [TypeSpec](https://typespec.io) [`@azure-tools/typespec-ts`](https://github.com/Azure/typespec-azure/tree/main/packages/typespec-ts) emitter.

Reviewer: ```suggestion
This library is primarily intended to be used in code generated by [AutoRest](https://github.com/Azure/Autorest) and [`autorest.typescript`](https://github.com/Azure/autorest.typescript).
```
